spec/docs/participants_spec.rb

```diff
@@ -0,0 +1,144 @@
+# frozen_string_literal: true
+
+require "swagger_helper"
+
+describe "API", type: :request, swagger_doc: "v1/api_spec.json", with_feature_flags: { participant_data_api: "active" } do
+  let(:token) { LeadProviderApiToken.create_with_random_token!(lead_provider: create(:lead_provider)) }
+  let(:bearer_token) { "Bearer #{token}" }
+  let(:Authorization) { bearer_token }
+
+  path "/api/v1/participants" do
+    get "Returns all participants" do
+      operationId :api_v1_participants_index
+      tags "participant"
+      produces "application/vnd.api+json"
+      security [bearerAuth: []]
+
+      parameter name: :filter,
+                in: :query,
+                schema: {
+                  type: :object,
+                  example: "",
+                  description: "This schema is used to search within collections to return more specific results.",
+                  properties: {
+                    updated_since: {
+                      description: "Return participants that have been updated since the date (ISO 8601 date format)",
+                      type: :string,
+                      example: "2021-05-13T11:21:55Z",
+                    },
+                  },
+                },
+                type: :object,
+                style: :deepObject,
+                explode: true,
+                required: false,
+                description: "Refine participants to return.",
+                example: { updated_since: "2020-11-13T11:21:55Z" }
```

Comment:
It would be really nice to find some way of reusing this stuff, although definitely not on this PR.

Reply:
Yeah, totally agree. I'd also like to commonise the pagination to the API controller, but will do that separately